# 异常处理

异常是干扰程序的正常流程的不寻常（但并非完全是出乎意料的）的事故。

当发现这样的事故时，你的程序应该抛出一个异常:

```javascript
const add = (a, b) => {
    // 此时只是抛出了一个错误，没有catch到
    if (typeof a !== 'number' || typeof b !== 'number') {
        throw {
            name: 'TypeError',
            message: 'add needs numbers'
        };
    }
    return a + b;
}
const try_it = () => {
    try {
        add('seven');
    } catch (e) {
        document.writeln(e.name + ': ' + e.message);
    }
}
try_it();// 正常执行程序会抛出一个错误，因为使用了catch方法所以错误被捕获了，不会中断余下代码的执行.
do something；// 养成这种好的习惯。
```

